pyrobosim/pyrobosim/gui/world_canvas.py

```diff
@@ -308,19 +286,24 @@ def show_planner_and_path(self, robot=None):
         while self.draw_lock:
             time.sleep(0.001)
         self.draw_lock = True
-        for artist in self.path_planner_artists:
-            artist.remove()
 
         color = robot.color if robot is not None else "m"
-
         if robot and robot.path_planner:
-            self.path_planner_artists = robot.path_planner.plot(
-                self.axes, path_color=color
-            )
-        elif self.world.path_planner:
-            self.path_planner_artists = self.world.path_planner.plot(
-                self.axes, path_color=color
-            )
+            path_planner_artists = robot.path_planner.plot(self.axes, path_color=color)
+            if "graph" in path_planner_artists:
+                for artist in self.path_planner_artists["graph"]:
+                    artist.remove()
```

Comment:
I think the reason switching between robots might be behaving unpredictably is that the artists should be removed always, regardless of whether the new thing to plot has `graph` or `path` defined.

Notice in lines 311-312 deleted, we were always removing everything first, and then adding if new things existed.

Same for the `artist.remove()` call in the path planner artists below.